project_singlefamilydetached/singlefamilydetached_with_upgrades.yml

```diff
@@ -0,0 +1,1165 @@
+        costs:
+          - value: 0.31
+            multiplier: Floor Area, Conditioned (ft^2)
+        lifetime: 999
+      - option: Infiltration|1.5 ACH50
+        apply_logic: Infiltration|2 ACH50
+        costs:
+          - value: 0.31
+            multiplier: Floor Area, Conditioned (ft^2)
+        lifetime: 999
+      - option: Mechanical Ventilation|Exhaust, 2010 ASHRAE 62.2
+        apply_logic: 
+          and:
+            - Mechanical Ventilation|None
+            - or:
+              - Infiltration|1.5 ACH50
+              - Infiltration|2 ACH50
+              - Infiltration|2.25 ACH50
+              - Infiltration|3 ACH50
+              - Infiltration|4 ACH50
+              - Infiltration|3.75 ACH50
+              - Infiltration|4 ACH50
+              - Infiltration|4.5 ACH50
+              - Infiltration|5 ACH50
+              - Infiltration|6 ACH50
+              - Infiltration|7 ACH50
+        costs: 
+          - value: 360
+            multiplier: Fixed (1)
+        lifetime: 999
+
+##Attic Insulation #s 02-04
+  #vented attics only, implemented via post processing
+  - upgrade_name: R-38 Attic Insulation
+    options:
+      - option: Insulation Unfinished Attic|Ceiling R-38, Vented
+        apply_logic: Insulation Unfinished Attic|Uninsulated, Vented
+        costs:
+          - value: 1.9
+            multiplier: Floor Area, Attic (ft^2)
+        lifetime: 999
+      - option: Insulation Unfinished Attic|Ceiling R-38, Vented
+        apply_logic: Insulation Unfinished Attic|Ceiling R-7, Vented
+        costs:
+          - value: 1.4
+            multiplier: Floor Area, Attic (ft^2)
+        lifetime: 999
+      - option: Insulation Unfinished Attic|Ceiling R-38, Vented
+        apply_logic: Insulation Unfinished Attic|Ceiling R-13, Vented
+        costs:
+          - value: 1.1
+            multiplier: Floor Area, Attic (ft^2)
+        lifetime: 999
+      - option: Insulation Unfinished Attic|Ceiling R-38, Vented
+        apply_logic: Insulation Unfinished Attic|Ceiling R-19, Vented
+        costs:
+          - value: 0.87
+            multiplier: Floor Area, Attic (ft^2)
+        lifetime: 999
+      - option: Insulation Unfinished Attic|Ceiling R-38, Vented
+        apply_logic: Insulation Unfinished Attic|Ceiling R-30, Vented
+        costs:
+          - value: 0.37
+            multiplier: Floor Area, Attic (ft^2)
+        lifetime: 999
+  - upgrade_name: R-49 Attic Insulation
+    options:
+      - option: Insulation Unfinished Attic|Ceiling R-49, Vented
+        apply_logic: Insulation Unfinished Attic|Uninsulated, Vented
+        costs:
+          - value: 2.4
+            multiplier: Floor Area, Attic (ft^2)
+        lifetime: 999
+        options:
+      - option: Insulation Unfinished Attic|Ceiling R-49, Vented
+        apply_logic: Insulation Unfinished Attic|Ceiling R-7, Vented
+        costs:
+          - value: 1.9
+            multiplier: Floor Area, Attic (ft^2)
+        lifetime: 999
+    options:
```

Comment:
```suggestion
```

Reply:
Oops. Good catch. I feel like the validator should've caught this...

@rajeee

Reply:
@joseph-robertson, oh! so are multiple options being put under a single upgrade?
My IDE does flag that as duplicate keys, however, yeah, the validation passes. I will check if we can make it fail the validation.

Reply:
Right. There were multiple `options:` under a single `upgrade_name:` but the validation still passed.